Add obsidian recipe

licht1stein opened the pull request:
### Brief summary of what the package does

Emacs front-end for [Obsidian Notes](https://obsidian.md).

Allows you to work with Obsidian Notes using Emacs. Obviously you already can open your Obsidian folder and start editing markdown files with Emacs. Obsidian.el adds conveniences like:

- autocomplete for tags
- `obsidian-insert-link` and `obsidian-insert-wikilink` commands with autcomplete
- `obsidian-follow-link-at-point` that correctly handles links generated by the Obsisian Notes app

Here's the announcement and reaction to the package on reddit:
https://www.reddit.com/r/emacs/comments/w82c8m/obsidianel_obsidian_notes_within_emacs/

### Direct link to the package repository

https://github.com/licht1stein/obsidian.el

### Your association with the package

Maintainer and author

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this!  Quick first pass --

`obsidian.el` with package-lint 20220607.738:
```
1 issue found:
9:120: warning: Use a non-snapshot version number for dependency on "elgrep" if possible.
```
I think you can use [1.0.0](https://stable.melpa.org/#/?q=elgrep) in this case.

`obsidian.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- obsidian.el#L206: Lighter should start with a space
- obsidian.el#L410: Loading a package should rarely add hooks
```
Especially in the case of line 410, this looks like it should be done within `define-globalized-minor-mode`, and then you would tell users to enable `global-obsidian-mode` (or whatever you call it) in their configs. 

In elisp, the conventional suffix for predicate functions is `-p` instead of `?`; in some cases this is actually [baked in](https://www.gnu.org/software/emacs/manual/html_node/cl/Type-Predicates.html).  It's better to be consistent with the community style, so you should aim to use `-p` where possible.

licht1stein commented:
Hi @riscy , thanks for the comments! I didn't know about global minor modes, this does look much better. I have addressed all of your comments, please take a look.

Also I noticed a strange thing, when running automatic linting on CI I get this warning on Emacs 28.1, although I don't get any warnings on 27 and 29:

https://github.com/licht1stein/obsidian.el/actions/runs/2812520229

It says:

`obsidian.el:411:52: Warning: ‘:group’ called as a function`

And it's regarding the global minor mode declaration:

```elisp
;;;###autoload
(define-globalized-minor-mode global-obsidian-mode obsidian-mode obsidian-enable-minor-mode
  :predicate '(markdown-mode)
  :group 'obsidian)
```

riscy commented:
Hmm, I have never seen that particular issue, but I haven't used `:predicate` before.  There may be some kind of a strange edge case being triggered in the define-globalized-minor-mode macro.

Here's a workaround that seems to work for me: add `(defgroup obsidian nil "<describe group>." :group 'text)` before your `defcustom`s.  Then the `define-globalized-minor-mode` will pick up the group and you don't need to specify it -- and as another small side effect, you also won't have to specify the `:group` in the defcustoms that follow in the file.

licht1stein commented:
@riscy thanks for the suggestion, I did that and it helped for Emacs 28. I still removed the :predicate form, cause it was introduced after Emacs 28, so it broke the Emacs 27 compatibility.

Everything is green now, I hope obsidian.el is good to  go :)

riscy commented:
Great, thanks!  Merging now.  And, welcome to MELPA. :)

licht1stein commented:
Wow thank you!
